pkg/testutil/testutil.go

```diff
@@ -17,15 +17,33 @@ package testutil
 import (
 	"net/url"
 	"runtime"
+	"strings"
 	"testing"
+	"time"
 )
 
-// WARNING: This is a hack.
-// Remove this when we are able to block/check the status of the go-routines.
-func ForceGosched() {
-	// possibility enough to sched up to 10 go routines.
-	for i := 0; i < 10000; i++ {
-		runtime.Gosched()
+func runnableGoroutineNumber() int {
```

Comment:
@jonboulle This is a too heavy solution in my opinion. We basically only wants to switch to another routine and let it run for a while... We do not need to wait until all routines to be blocked.